idflakies-maven-plugin/src/main/java/edu/illinois/cs/dt/tools/plugin/IncDetectorMojo.java

```diff
@@ -0,0 +1,651 @@
+    }
+
+    private static boolean inImmutableList(Field field) {
+        boolean isFinal = false;
+        if (Modifier.isFinal(field.getModifiers())) {
+            isFinal = true;
+        }
+
+        if ((field.getType().isPrimitive() || field.getDeclaringClass().isEnum()) && isFinal) {
+            return true;
+        }
+
+        for (String immutableTypeName: immutableList) {
+            if ((field.getType().getName().equals(immutableTypeName)) && isFinal) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static BufferedWriter getWriter(String filePath) {
+        Path path = Paths.get(filePath);
+        BufferedWriter writer = null;
+        try {
+            if (path.getParent() != null && !Files.exists(path.getParent())) {
+                Files.createDirectories(path.getParent());
+            }
+            writer = Files.newBufferedWriter(path, StandardCharsets.UTF_8);
+        } catch (IOException e1) {
+            e1.printStackTrace();
+        }
+        return writer;
+    }
+
+    public static void writeJarChecksums(List<String> sfPathString, String artifactsDir, List<Pair> jarCheckSums) {
+        String outFilename = Paths.get(artifactsDir, JAR_CHECKSUMS).toString();
+        try (BufferedWriter writer = getWriter(outFilename)) {
+            if (jarCheckSums != null) {
+                // we already computed the checksums while checking with old version in RunMojo#hasSameJarChecksum()
+                for (Pair<String, String> pair : jarCheckSums) {
+                    writer.write(pair.toString());
+                    writer.write(System.lineSeparator());
+                }
+            } else {
+                for (String path : sfPathString) {
+                    if (path.isEmpty()) {
+                        continue;
+                    }
+                    writer.write(getJarToChecksumMapping(path).toString());
+                    writer.write(System.lineSeparator());
+                }
+            }
+        } catch (IOException ioe) {
+            ioe.printStackTrace();
+        }
+    }
+
+    public static void writeClassPath(String sfPathString, String artifactsDir) {
+        String outFilename = Paths.get(artifactsDir, SF_CLASSPATH).toString();
+        try (BufferedWriter writer = getWriter(outFilename)) {
+            writer.write(sfPathString + System.lineSeparator());
+        } catch (IOException ioe) {
+            ioe.printStackTrace();
+        }
+    }
+
+    public static String pathToString(List<String> classPath) {
+        StringBuilder sb = new StringBuilder();
+        Iterator<String> iterator = classPath.iterator();
+        while (iterator.hasNext()) {
+            sb.append(iterator.next());
+            if (iterator.hasNext()) {
+                sb.append(File.pathSeparator);
+            }
+        }
+        return sb.toString();
+    }
+
+    public static String millsToSeconds(long value) {
```

Comment:
If you simplify that logic so there is no longer a divide between Ekstazi and STARTS, then don't need this method, can just inline into that one location.